tests/x11/rmt/rmt_feature.pm

```diff
@@ -0,0 +1,98 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Add rmt configuration test and basic configuration via
+#    rmt-wizard, test enable/disable products/repo, test rmt sync
+#    rmt mirror, test import SMT data to RMT
+# Maintainer: Yutao wang <yuwang@suse.com>
+
+use strict;
+use warnings;
+use testapi;
+use base 'x11test';
+use repo_tools;
+use utils;
+use x11utils 'turn_off_gnome_screensaver';
+
+sub run {
+    x11_start_program('xterm -geometry 150x35+5+5', target_match => 'xterm');
+    # Avoid blank screen since smt sync needs time
+    turn_off_gnome_screensaver;
+    become_root;
+    rmt_wizard();
+    # sync from SCC
+    rmt_sync;
+    # enable all modules of products at one arch
+    my $pro = get_var("PRODUCT_ALLM");
+    assert_script_run("rmt-cli p e $pro --all-modules");
+    assert_script_run("rmt-cli product list | grep $pro");
+    # enable all modules of products at four arch
+    my $pro1 = get_var("PRODUCT_ALLMA");
+    assert_script_run("rmt-cli products enable --all-modules $pro1");
+    assert_script_run("rmt-cli product list | grep " . $pro1 . "/aarch64");
+    assert_script_run("rmt-cli product list | grep " . $pro1 . "/ppc64le");
+    assert_script_run("rmt-cli product list | grep " . $pro1 . "/s390x");
+    assert_script_run("rmt-cli product list | grep " . $pro1 . "/x86_64");
+    my @proid = split(/\n/, script_output("rmt-cli products list | awk -F '|' '{print \$2}'"));
+    diag "my output: @proid";
+    foreach my $i (@proid) {
+        script_run("rmt-cli products disable $i") if ($i =~ /\d{4}/);
+    }
+
+    # enable product with product ID 1798-Web and Scripting Module/15.1/x86_64
+    # 1973-Web and Scripting Module/15.2/aarch64 1974-Web and Scripting Module/15.2/ppc64le
+    assert_script_run("rmt-cli product enable 1798 1973 1974");
```

Comment:
Is there any pre-condition to do this? Need mirror all products?

Reply:
Random ID